sdk/tables/azure-data-tables/azure/data/tables/_decoder.py

```diff
@@ -3,188 +3,175 @@
-    _service_value: str
+def _no_op(value: Any) -> Any:
+    return value
 
-    @property
-    def tables_service_value(self) -> str:
-        try:
-            return self._service_value
-        except AttributeError:
-            return ""
 
-
-NO_ODATA = {
-    int: EdmType.INT32,
-    str: EdmType.STRING,
-    bool: EdmType.BOOLEAN,
-    float: EdmType.DOUBLE,
-}
+def _none_op(_: Any) -> None:
+    return None
 
 
 class TableEntityDecoder:
-    def __init__(
-        self,
-        *,
-        flatten_result_entity: bool = False,
-        convert_map: Optional[DecoderMapType] = None,
-    ) -> None:
-        self.convert_map = convert_map
-        self.flatten_result_entity = flatten_result_entity
+    def __init__(self, *, convert_map: DecoderMapType, entity_format: Any) -> None:
+        self._edm_types: Dict[EdmType, DecodeCallable] = {
+            EdmType.BINARY: b64decode,
+            EdmType.INT32: int,
+            EdmType.INT64: self._decode_int64,
+            EdmType.DOUBLE: float,
+            EdmType.DATETIME: deserialize_iso,
+            EdmType.GUID: UUID,
+            EdmType.STRING: str,
+            EdmType.BOOLEAN: self._decode_boolean,
+        }
+        self._decode_types: Dict[Type, Union[EdmType, DecodeCallable]] = {
+            str: EdmType.STRING,
+            int: EdmType.INT32,
+            bool: EdmType.BOOLEAN,
+            datetime: EdmType.DATETIME,
+            float: EdmType.DOUBLE,
+            UUID: EdmType.GUID,
+            bytes: EdmType.BINARY,
+        }
+        for key, value in convert_map.items():
+            if isinstance(key, EdmType):
+                self._edm_types[key] = value
+            else:
+                self._decode_types[key] = value
+        self._property_types: Dict[str, Union[EdmType, DecodeCallable]] = {
+            "PartitionKey": _no_op,
+            "RowKey": _no_op,
+            "Timestamp": _none_op,
+        }
+        if entity_format:
+            property_name: str
+            property_type: Union[Type, EdmType]
+            if is_typeddict(entity_format):
+                # Scrape type encoding from typeddict defintion
+                for property_name, property_type in _annotations(entity_format).items():
+                    property_type = _get_annotation_type(property_type)
+                    self._add_property_type(property_name, property_type)
+            elif is_dataclass(entity_format):
+                # Scrape type decoding from a dataclass definition
+                for entity_field in fields(entity_format):
+                    property_type = _get_annotation_type(entity_field.type)
+                    self._add_property_type(entity_field.name, property_type)
+            else:
+                # Finally, we'll check if it's a simply dict of properties to types.
+                try:
+                    for property_name, property_type in entity_format.items():
+                        property_type = _get_annotation_type(property_type)
```

Comment:
do we want to change `property_type` on 88 to `annotation_type` or can they end up being same type after `_get_annotation_type`